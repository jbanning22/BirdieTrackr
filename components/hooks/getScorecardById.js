import axios from 'axios';
import {useQuery} from '@tanstack/react-query';
import AsyncStorage from '@react-native-async-storage/async-storage';

export const useGetScorecardById = id => {
  const getScorecardByIdURL = `http://ec2-54-173-139-185.compute-1.amazonaws.com:3000/scorecard/${id}`;

  const getScorecard = async () => {
    const token = await AsyncStorage.getItem('token');
    const headers = {
      Authorization: `Bearer ${token}`,
    };
    const response = await axios.get(getScorecardByIdURL, {
      headers,
    });
    return response ? response.data : null;
  };

  const {data, isLoading, isError, error} = useQuery(
    ['scorecardByIdData'],
    getScorecard,
  );
  return {data, isLoading, isError, error};
};
