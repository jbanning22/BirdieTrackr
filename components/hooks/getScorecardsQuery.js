import axios from 'axios';
import {useQuery} from '@tanstack/react-query';
import AsyncStorage from '@react-native-async-storage/async-storage';

const getScorecardsUrl =
  'http://ec2-54-87-189-240.compute-1.amazonaws.com:3000/scorecard';

const getScorecards = async () => {
  const token = await AsyncStorage.getItem('token');
  const headers = {
    Authorization: `Bearer ${token}`,
  };
  const response = await axios.get(getScorecardsUrl, {
    headers,
  });
  return response ? response.data : null;
};

export const useGetScorecards = () => {
  const {data, isLoading, isError, error} = useQuery(
    ['scorecardData'],
    getScorecards,
  );
  return {data, isLoading, isError, error};
};
