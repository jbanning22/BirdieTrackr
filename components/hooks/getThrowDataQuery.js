import axios from 'axios';
import {useQuery} from '@tanstack/react-query';
import AsyncStorage from '@react-native-async-storage/async-storage';

const getThrowsUrl =
  'http://ec2-54-87-189-240.compute-1.amazonaws.com:3000/measure-throws';

const getThrows = async () => {
  const token = await AsyncStorage.getItem('token');
  const headers = {
    Authorization: `Bearer ${token}`,
  };
  try {
    const response = await axios.get(getThrowsUrl, {headers});
    return response ? response.data : null;
  } catch (error) {
    throw error;
  }
};

export const useGetThrowDetails = () => {
  const {data, isLoading, isError, error} = useQuery(['throwData'], getThrows);
  return {data, isLoading, isError, error};
};
