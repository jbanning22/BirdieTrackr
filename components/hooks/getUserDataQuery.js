import axios from 'axios';
import {useQuery} from '@tanstack/react-query';
import AsyncStorage from '@react-native-async-storage/async-storage';

const getUserUrl =
  'http://ec2-54-87-189-240.compute-1.amazonaws.com:3000/users/me';

const getUser = async () => {
  const token = await AsyncStorage.getItem('token');
  const headers = {
    Authorization: `Bearer ${token}`,
  };
  try {
    const response = await axios.get(getUserUrl, {headers});
    return response.data;
  } catch (error) {
    throw error;
  }
};

export const useGetUserDetails = () => {
  const {data, isLoading, isError, error} = useQuery(['userData'], getUser);
  return {data, isLoading, isError, error};
};
