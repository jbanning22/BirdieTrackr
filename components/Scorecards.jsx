import {
  StyleSheet,
  Text,
  View,
  FlatList,
  TouchableOpacity,
  ActivityIndicator,
  ImageBackground,
} from 'react-native';
import React from 'react';
import {Alert} from 'react-native';
import axios from 'axios';
import AsyncStorage from '@react-native-async-storage/async-storage';
import moment from 'moment';
import {SafeAreaView} from 'react-native-safe-area-context';
import {FontAwesomeIcon} from '@fortawesome/react-native-fontawesome';
import {faCheck} from '@fortawesome/free-solid-svg-icons/faCheck';
import {faExclamation} from '@fortawesome/free-solid-svg-icons/faExclamation';
import {faTrashCan} from '@fortawesome/free-solid-svg-icons/faTrashCan';
import {useQueryClient} from '@tanstack/react-query';
import {useGetScorecards} from './hooks/getScorecardsQuery';
import {useGetUserDetails} from './hooks/getUserDataQuery';
import myImage from '../assets/images/BasketBackground2.png';
import {Dimensions} from 'react-native';
import LargeButton from './button/LargeButton';

const Scorecards = ({navigation}) => {
  const {
    data: scorecardData,
    // isLoading: isScorecardLoading,
    // isError: isScorecardError,
    // error: Scorecarderror,
  } = useGetScorecards();
  const {
    data: userDetails,
    isLoading: isUserDataLoading,
    // isError: isUserDataError,
    // error: UserDataerror,
  } = useGetUserDetails();
  const queryClient = useQueryClient();
  const windowWidth = Dimensions.get('window').width;
  // const windowHeight = Dimensions.get('window').height;

  const handleScorecardPressed = async id => {
    const token = await AsyncStorage.getItem('token');
    const headers = {
      Authorization: `Bearer ${token}`,
    };
    try {
      const scorecard = await axios.get(
        `http://ec2-54-173-139-185.compute-1.amazonaws.com:3000/scorecard/${id}`,
        {
          headers,
        },
      );
      await navigation.navigate('FullScorecard', {
        id: scorecard.data.id,
      });
    } catch (error) {
      // eslint-disable-next-line no-console
      console.log(error);
    }
  };

  const renderItem = ({item}) => {
    const date = moment(item.createdAt).format('MMMM Do, YYYY');
    // console.log('item in flatlist is: ', item);
    const score = course => {
      const totalStrokes = course.holes.reduce(
        (acc, hole) => acc + hole.strokes,
        0,
      );
      const totalPars = course.holes.reduce((acc, hole) => acc + hole.par, 0);
      return totalStrokes - totalPars;
    };

    const icon = item.isCompleted ? (
      <FontAwesomeIcon
        icon={faCheck}
        color={'white'}
        size={12}
        style={{margin: 8}}
      />
    ) : (
      <FontAwesomeIcon
        icon={faExclamation}
        color={'white'}
        size={12}
        style={{margin: 8}}
      />
    );

    return (
      <View style={styles.flatListItemContainer}>
        <TouchableOpacity
          onPress={() => handleScorecardPressed(item.id, item.courseLength)}>
          <View style={styles.flatlistTextItemStyle}>
            <Text style={styles.renderCourseName}>{item.courseName}</Text>
            <Text style={styles.renderHoleText}>{item.courseLength} Holes</Text>
            <Text style={styles.renderText}>{date}</Text>
          </View>
        </TouchableOpacity>

        <View
          style={{
            flexDirection: 'column',
            justifyContent: 'space-between',
            marginRight: 5,
          }}>
          {icon}
          <Text style={styles.renderScore}>{score(item)}</Text>
          <TouchableOpacity onPress={() => deleteScorecard(item.id)}>
            <FontAwesomeIcon
              icon={faTrashCan}
              color={'white'}
              size={12}
              style={{margin: 8}}
            />
          </TouchableOpacity>
        </View>
      </View>
    );
  };

  const deleteScorecard = async id => {
    const token = await AsyncStorage.getItem('token');
    const headers = {
      Authorization: `Bearer ${token}`,
    };
    Alert.alert(
      'Delete Scorecard',
      'Are you sure you want to delete this scorecard?',
      [
        {text: 'Cancel', style: 'cancel'},
        {
          text: 'Delete',
          onPress: async () => {
            try {
              await axios.delete(
                `http://ec2-54-173-139-185.compute-1.amazonaws.com:3000/scorecard/${id}`,
                {headers},
              );
              queryClient.invalidateQueries('scorecardData');
            } catch (error) {
              // eslint-disable-next-line no-console
              console.log(error);
            }
          },
        },
      ],
      {cancelable: false},
    );
  };

  return (
    <SafeAreaView style={styles.box1}>
      <ImageBackground source={myImage} style={styles.imageBackground}>
        <View
          style={{
            padding: 10,
            alignItems: 'flex-start',
            backgroundColor: 'white',
          }}>
          <Text
            style={{
              fontFamily: 'Satoshi-Medium',
              fontSize: 20,
              marginTop: 10,
            }}>
            Hi,{' '}
            {isUserDataLoading ? (
              <ActivityIndicator></ActivityIndicator>
            ) : (
              userDetails.firstName
            )}
          </Text>
          <Text style={{flexDirection: 'column', color: 'grey'}}>
            Welcome back to{' '}
            <Text
              style={{
                color: '#45B369',
                fontSize: 14,
                fontFamily: 'Satoshi-Medium',
              }}>
              DG Scorecard
            </Text>
            !
          </Text>
        </View>

        <View
          style={{
            padding: 20,
            alignItems: 'center',
            alignContent: 'center',
            justifyContent: 'center',
            height: 570,
            width: windowWidth,
          }}>
          {scorecardData && scorecardData.length === 0 ? (
            <Text style={{fontSize: 18, fontWeight: '500'}}>
              You have not recorded any rounds yet.
            </Text>
          ) : (
            <FlatList
              renderItem={renderItem}
              data={scorecardData}
              showsVerticalScrollIndicator={false}
              keyExtractor={item => item.id.toString()}
            />
          )}
        </View>

        <View
          style={{
            alignItems: 'center',
            marginBottom: 20,
          }}>
          <LargeButton
            buttonText="Create Scorecard"
            onPress={() => navigation.navigate('CreateScorecard')}
          />
        </View>
      </ImageBackground>
    </SafeAreaView>
  );
};

export default Scorecards;

const styles = StyleSheet.create({
  imageBackground: {
    resizeMode: 'cover',
  },
  box1: {
    flex: 1,
    // alignItems: 'center',
    justifyContent: 'space-between',
    backgroundColor: 'white',
  },
  flatListItemContainer: {
    flexDirection: 'row',
    backgroundColor: '#2D6061',
    borderRadius: 10,
    marginBottom: 10,
  },
  homeText: {
    fontSize: 40,
    fontWeight: '400',
    fontFamily: 'Satoshi-Medium',
    marginBottom: 15,
    color: 'black',
  },
  createScorecardButton: {
    width: 327,
    height: 56,
    justifyContent: 'center',
    alignItems: 'center',
    alignSelf: 'center',
    backgroundColor: '#2D6061',
    borderRadius: 14,
  },
  buttonText: {
    fontFamily: 'Satoshi-Medium',
    color: 'white',
    fontSize: 18,
    fontWeight: '500',
  },
  flatlistTextItemStyle: {
    width: 280,
    height: 80,
    margin: 15,
    alignItems: 'flex-start',
  },
  renderItemStyle: {
    flexDirection: 'column',
  },
  renderCourseName: {
    fontFamily: 'Satoshi-Medium',
    fontSize: 20,
    fontWeight: '700',
    color: 'white',
    textAlign: 'left',
    marginBottom: 5,
  },
  renderHoleText: {
    fontFamily: 'Satoshi-Medium',
    fontSize: 16,
    fontWeight: '500',
    color: 'white',
    textAlign: 'left',
    marginBottom: 2,
  },
  renderScore: {
    fontFamily: 'Satoshi-Medium',
    fontSize: 24,
    fontWeight: '500',
    color: 'white',
    alignSelf: 'flex-end',
    margin: 5,
  },
  renderText: {
    fontFamily: 'Satoshi-Medium',
    fontSize: 12,
    fontWeight: '500',
    color: 'white',
  },
});
