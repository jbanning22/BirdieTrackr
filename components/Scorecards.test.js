import React from 'react';
import {
  fireEvent,
  render,
  screen,
  waitFor,
} from '@testing-library/react-native';
import Scorecards from './Scorecards';
import axios from 'axios';
import {QueryClient, QueryClientProvider} from '@tanstack/react-query';
import {useGetScorecards} from './hooks/getScorecardsQuery';
import AsyncStorage from '@react-native-async-storage/async-storage';

jest.mock('@fortawesome/react-native-fontawesome', () => ({
  FontAwesomeIcon: () => null,
}));
jest.mock('@react-native-async-storage/async-storage');
jest.mock('./hooks/getScorecardsQuery', () => ({
  useGetScorecards: jest.fn(),
}));
jest.mock('axios');

describe('Scorecards Landing', () => {
  let queryClient;
  const mockedData = [
    {
      id: 3,
      createdAt: '2023-05-31T19:14:39.335Z',
      isCompleted: false,
      courseLength: 18,
      courseName: 'Test',
      playerId: 1,
      holes: [
        {
          id: 37,
          holeNumber: 1,
          par: 3,
          strokes: 0,
          playerId: 1,
          scorecardId: 3,
        },
        {
          id: 38,
          holeNumber: 2,
          par: 3,
          strokes: 0,
          playerId: 1,
          scorecardId: 3,
        },
        {
          id: 39,
          holeNumber: 3,
          par: 3,
          strokes: 0,
          playerId: 1,
          scorecardId: 3,
        },
        {
          id: 40,
          holeNumber: 4,
          par: 3,
          strokes: 0,
          playerId: 1,
          scorecardId: 3,
        },
        {
          id: 41,
          holeNumber: 5,
          par: 3,
          strokes: 0,
          playerId: 1,
          scorecardId: 3,
        },
        {
          id: 42,
          holeNumber: 6,
          par: 3,
          strokes: 0,
          playerId: 1,
          scorecardId: 3,
        },
        {
          id: 43,
          holeNumber: 7,
          par: 3,
          strokes: 0,
          playerId: 1,
          scorecardId: 3,
        },
        {
          id: 44,
          holeNumber: 8,
          par: 3,
          strokes: 0,
          playerId: 1,
          scorecardId: 3,
        },
        {
          id: 45,
          holeNumber: 9,
          par: 3,
          strokes: 0,
          playerId: 1,
          scorecardId: 3,
        },
        {
          id: 46,
          holeNumber: 10,
          par: 3,
          strokes: 0,
          playerId: 1,
          scorecardId: 3,
        },
        {
          id: 47,
          holeNumber: 11,
          par: 3,
          strokes: 0,
          playerId: 1,
          scorecardId: 3,
        },
        {
          id: 48,
          holeNumber: 12,
          par: 3,
          strokes: 0,
          playerId: 1,
          scorecardId: 3,
        },
        {
          id: 49,
          holeNumber: 13,
          par: 3,
          strokes: 0,
          playerId: 1,
          scorecardId: 3,
        },
        {
          id: 50,
          holeNumber: 14,
          par: 3,
          strokes: 0,
          playerId: 1,
          scorecardId: 3,
        },
        {
          id: 51,
          holeNumber: 15,
          par: 3,
          strokes: 0,
          playerId: 1,
          scorecardId: 3,
        },
        {
          id: 52,
          holeNumber: 16,
          par: 3,
          strokes: 0,
          playerId: 1,
          scorecardId: 3,
        },
        {
          id: 53,
          holeNumber: 17,
          par: 3,
          strokes: 0,
          playerId: 1,
          scorecardId: 3,
        },
        {
          id: 54,
          holeNumber: 18,
          par: 3,
          strokes: 0,
          playerId: 1,
          scorecardId: 3,
        },
      ],
    },
  ];
  beforeEach(() => {
    queryClient = new QueryClient();
    AsyncStorage.getItem.mockResolvedValue('dummy_token');
    useGetScorecards.mockReturnValue({
      data: mockedData,
      isLoading: false,
      isError: false,
      error: null,
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot', () => {
    const {toJSON} = render(
      <QueryClientProvider client={queryClient}>
        <Scorecards />,
      </QueryClientProvider>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('should display screen title', async () => {
    render(
      <QueryClientProvider client={queryClient}>
        <Scorecards />,
      </QueryClientProvider>,
    );
    await waitFor(() => {
      expect(screen.getByText('Scorecards')).toBeTruthy();
    });
  });

  it('should display create scorecard button', async () => {
    render(
      <QueryClientProvider client={queryClient}>
        <Scorecards />,
      </QueryClientProvider>,
    );
    await waitFor(() => {
      expect(screen.getByText('Create Scorecard')).toBeTruthy();
    });
  });

  it('should navigate to create scorecard screen', async () => {
    const navigation = {
      navigate: jest.fn(),
    };
    render(
      <QueryClientProvider client={queryClient}>
        <Scorecards navigation={navigation} />,
      </QueryClientProvider>,
    );
    fireEvent.press(screen.getByText('Create Scorecard'));
    await waitFor(() => {
      expect(navigation.navigate).toHaveBeenCalledWith('CreateScorecard');
    });
  });

  it('should display scorecards when data is fetched', async () => {
    render(
      <QueryClientProvider client={queryClient}>
        <Scorecards />
      </QueryClientProvider>,
    );
    await waitFor(() => {
      expect(screen.findByText('Test')).toBeTruthy();
    });
  });
});
