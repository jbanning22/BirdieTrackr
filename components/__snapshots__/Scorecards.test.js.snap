// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scorecards Landing should match the snapshot 1`] = `
[
  <RNCSafeAreaView
    edges={
      {
        "bottom": "additive",
        "left": "additive",
        "right": "additive",
        "top": "additive",
      }
    }
    style={
      {
        "backgroundColor": "white",
        "flex": 1,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        {
          "resizeMode": "cover",
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/images/BasketBackground2.png",
          }
        }
        style={
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            {
              "height": undefined,
              "width": undefined,
            },
            undefined,
          ]
        }
      />
      <View
        style={
          {
            "alignItems": "flex-start",
            "backgroundColor": "white",
            "padding": 10,
          }
        }
      >
        <Text
          style={
            {
              "fontFamily": "Satoshi-Medium",
              "fontSize": 20,
              "marginTop": 10,
            }
          }
        >
          Hi,
           
          <ActivityIndicator />
        </Text>
        <Text
          style={
            {
              "color": "grey",
              "flexDirection": "column",
            }
          }
        >
          Welcome back to
           
          <Text
            style={
              {
                "color": "#45B369",
                "fontFamily": "Satoshi-Medium",
                "fontSize": 14,
              }
            }
          >
            BirdieTrackr
          </Text>
          !
        </Text>
      </View>
      <View
        style={
          {
            "alignContent": "center",
            "alignItems": "center",
            "height": 570,
            "justifyContent": "center",
            "padding": 20,
            "width": 750,
          }
        }
      >
        <RCTScrollView
          data={
            [
              {
                "courseLength": 18,
                "courseName": "Test",
                "createdAt": "2023-05-31T19:14:39.335Z",
                "holes": [
                  {
                    "holeNumber": 1,
                    "id": 37,
                    "par": 3,
                    "playerId": 1,
                    "scorecardId": 3,
                    "strokes": 0,
                  },
                  {
                    "holeNumber": 2,
                    "id": 38,
                    "par": 3,
                    "playerId": 1,
                    "scorecardId": 3,
                    "strokes": 0,
                  },
                  {
                    "holeNumber": 3,
                    "id": 39,
                    "par": 3,
                    "playerId": 1,
                    "scorecardId": 3,
                    "strokes": 0,
                  },
                  {
                    "holeNumber": 4,
                    "id": 40,
                    "par": 3,
                    "playerId": 1,
                    "scorecardId": 3,
                    "strokes": 0,
                  },
                  {
                    "holeNumber": 5,
                    "id": 41,
                    "par": 3,
                    "playerId": 1,
                    "scorecardId": 3,
                    "strokes": 0,
                  },
                  {
                    "holeNumber": 6,
                    "id": 42,
                    "par": 3,
                    "playerId": 1,
                    "scorecardId": 3,
                    "strokes": 0,
                  },
                  {
                    "holeNumber": 7,
                    "id": 43,
                    "par": 3,
                    "playerId": 1,
                    "scorecardId": 3,
                    "strokes": 0,
                  },
                  {
                    "holeNumber": 8,
                    "id": 44,
                    "par": 3,
                    "playerId": 1,
                    "scorecardId": 3,
                    "strokes": 0,
                  },
                  {
                    "holeNumber": 9,
                    "id": 45,
                    "par": 3,
                    "playerId": 1,
                    "scorecardId": 3,
                    "strokes": 0,
                  },
                  {
                    "holeNumber": 10,
                    "id": 46,
                    "par": 3,
                    "playerId": 1,
                    "scorecardId": 3,
                    "strokes": 0,
                  },
                  {
                    "holeNumber": 11,
                    "id": 47,
                    "par": 3,
                    "playerId": 1,
                    "scorecardId": 3,
                    "strokes": 0,
                  },
                  {
                    "holeNumber": 12,
                    "id": 48,
                    "par": 3,
                    "playerId": 1,
                    "scorecardId": 3,
                    "strokes": 0,
                  },
                  {
                    "holeNumber": 13,
                    "id": 49,
                    "par": 3,
                    "playerId": 1,
                    "scorecardId": 3,
                    "strokes": 0,
                  },
                  {
                    "holeNumber": 14,
                    "id": 50,
                    "par": 3,
                    "playerId": 1,
                    "scorecardId": 3,
                    "strokes": 0,
                  },
                  {
                    "holeNumber": 15,
                    "id": 51,
                    "par": 3,
                    "playerId": 1,
                    "scorecardId": 3,
                    "strokes": 0,
                  },
                  {
                    "holeNumber": 16,
                    "id": 52,
                    "par": 3,
                    "playerId": 1,
                    "scorecardId": 3,
                    "strokes": 0,
                  },
                  {
                    "holeNumber": 17,
                    "id": 53,
                    "par": 3,
                    "playerId": 1,
                    "scorecardId": 3,
                    "strokes": 0,
                  },
                  {
                    "holeNumber": 18,
                    "id": 54,
                    "par": 3,
                    "playerId": 1,
                    "scorecardId": 3,
                    "strokes": 0,
                  },
                ],
                "id": 3,
                "isCompleted": false,
                "playerId": 1,
              },
            ]
          }
          getItem={[Function]}
          getItemCount={[Function]}
          keyExtractor={[Function]}
          onContentSizeChange={[Function]}
          onLayout={[Function]}
          onMomentumScrollBegin={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          showsVerticalScrollIndicator={false}
          stickyHeaderIndices={[]}
          viewabilityConfigCallbackPairs={[]}
        >
          <View>
            <View
              onFocusCapture={[Function]}
              onLayout={[Function]}
              style={null}
            >
              <View
                style={
                  {
                    "backgroundColor": "#2D6061",
                    "borderRadius": 10,
                    "flexDirection": "row",
                    "marginBottom": 10,
                  }
                }
              >
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      {
                        "alignItems": "flex-start",
                        "height": 80,
                        "margin": 15,
                        "width": 280,
                      }
                    }
                  >
                    <Text
                      style={
                        {
                          "color": "white",
                          "fontFamily": "Satoshi-Medium",
                          "fontSize": 20,
                          "fontWeight": "700",
                          "marginBottom": 5,
                          "textAlign": "left",
                        }
                      }
                    >
                      Test
                    </Text>
                    <Text
                      style={
                        {
                          "color": "white",
                          "fontFamily": "Satoshi-Medium",
                          "fontSize": 16,
                          "fontWeight": "500",
                          "marginBottom": 2,
                          "textAlign": "left",
                        }
                      }
                    >
                      18
                       Holes
                    </Text>
                    <Text
                      style={
                        {
                          "color": "white",
                          "fontFamily": "Satoshi-Medium",
                          "fontSize": 12,
                          "fontWeight": "500",
                        }
                      }
                    >
                      May 31st, 2023
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    {
                      "flexDirection": "column",
                      "justifyContent": "space-between",
                      "marginRight": 5,
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "alignSelf": "flex-end",
                        "color": "white",
                        "fontFamily": "Satoshi-Medium",
                        "fontSize": 24,
                        "fontWeight": "500",
                      }
                    }
                  >
                    -54
                  </Text>
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "opacity": 1,
                      }
                    }
                  />
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "marginBottom": 20,
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "#2D6061",
              "borderRadius": 14,
              "height": 56,
              "justifyContent": "center",
              "margin": 15,
              "opacity": 1,
              "width": 327,
            }
          }
        >
          <Text
            style={
              {
                "color": "white",
                "fontFamily": "Satoshi-Medium",
                "fontSize": 18,
                "fontWeight": "500",
              }
            }
          >
            Create Scorecard
          </Text>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>,
  ",",
]
`;
